packages/mobile/src/web3/saga.ts

```diff
@@ -133,9 +122,15 @@ function* checkSyncProgressWorker() {
           continue
         }
 
-        Logger.debug(TAG, 'Sync Progress Completed')
-        yield put(setSyncProgress(100))
-        yield put(setIsReady(true))
+        const latestBlock: Block = yield getLatestBlock()
```

Comment:
i used to see instances where web3 CLAIMED to be done syncing but actually latestBlock was still 0, which is why we used to check for that case and keep waiting.